service.py

```diff
@@ -41,6 +41,7 @@ def fail(self, msg):
     def do_GET(self):
         url = urlparse(self.path)
         qs = parse_qs(url.query)
+        version = qs.get('version')
```

Comment:
You could do something like `version = qs.get('version', "2")` here. 
This means if version is not found, then it defaults to "2". 
This will simply the code at L53 (you will no longer need the `version and` part of `version and version[0] == "3"`